library/trans.c

```diff
@@ -442,41 +444,54 @@ void dnet_trans_clean_list(struct list_head *head, int error)
 
 void dnet_update_stall_backend_weights(struct list_head *stall_transactions)
 {
-	struct dnet_trans *t, *tmp;
+	struct dnet_io_req *r, *tmp;
+	struct dnet_cmd *cmd;
 	struct dnet_net_state *st;
 	double old_cache_weight, new_cache_weight;
 	double old_disk_weight, new_disk_weight;
 	int err;
 
-	list_for_each_entry_safe(t, tmp, stall_transactions, trans_list_entry) {
-		st = t->st;
+	list_for_each_entry_safe(r, tmp, stall_transactions, req_entry) {
+		cmd = r->header;
+		st = r->st;
 
-		err = dnet_get_backend_weight(st, t->cmd.backend_id, DNET_IO_FLAGS_CACHE, &old_cache_weight);
+		err = dnet_get_backend_weight(st, cmd->backend_id, DNET_IO_FLAGS_CACHE, &old_cache_weight);
 		if (!err) {
 			new_cache_weight = old_cache_weight;
 			if (new_cache_weight >= 2) {
 				new_cache_weight /= 10;
-				dnet_set_backend_weight(st, t->cmd.backend_id, DNET_IO_FLAGS_CACHE, new_cache_weight);
+				dnet_set_backend_weight(st, cmd->backend_id, DNET_IO_FLAGS_CACHE, new_cache_weight);
 			}
 
-			err = dnet_get_backend_weight(st, t->cmd.backend_id, 0, &old_disk_weight);
+			err = dnet_get_backend_weight(st, cmd->backend_id, 0, &old_disk_weight);
 			if (!err) {
 				new_disk_weight = old_disk_weight;
 				if (new_disk_weight >= 2) {
 					new_disk_weight /= 10;
-					dnet_set_backend_weight(st, t->cmd.backend_id, 0, new_disk_weight);
+					dnet_set_backend_weight(st, cmd->backend_id, 0, new_disk_weight);
 				}
 
 				dnet_log(st->n, DNET_LOG_INFO, "%s/%d: TIMEOUT: update backend weight: weight: "
 						"cache: %f -> %f, disk: %f -> %f",
-					 dnet_state_dump_addr(st), t->cmd.backend_id,
+					 dnet_state_dump_addr(st), cmd->backend_id,
 					 old_cache_weight, new_cache_weight,
 					 old_disk_weight, new_disk_weight);
 			}
 		}
 	}
 }
 
+int dnet_timespec_less(struct timespec *l, struct timespec *r)
```

Comment:
How about making and using more general variant?
```suggestion
static int dnet_timespec_cmp(struct timespec *l, struct timespec *r)
```

Reply:
`static` ok. Are you sure about `_cmp`? Functions with such a postfix usually return values -1, 0, 1 (less, eq, greater).

Reply:
Ok, I understand, you meant general variant.